docs/controls/controls.rst

```diff
@@ -0,0 +1,477 @@
+
+In the actual file dialog element, we have to change how the ``folder`` property is set, and ensure that the ``onAccepted`` handler uses the ``file`` property instead of the ``fileUrl`` property. Apart from these details, the usage is identical to the ``FileDialog`` from ``QtQuick.Dialogs``.
+
+.. literalinclude:: src/imageviewer-native/main.qml
+    :lines: 1-10, 61-71, 84-
+
+In addition to the QML changes, we also need to alter the project file of the image viewer to include the ``widgets`` module.
+
+.. literalinclude:: src/imageviewer-native/imageviewer-native.pro
+    :lines: 1
+
+And we need to update ``main.qml`` to instantiate a ``QApplication`` object instead of a ``QGuiApplication`` object. This is because the ``QGuiApplication`` class contains the minimal environment needed for a graphical application, while ``QApplication`` extends ``QGuiApplication`` with features needed to support ``QtWidgets``.
+    
+.. literalinclude:: src/imageviewer-native/main.cpp
+    :language: cpp
+    :lines: 1, 4-8, 11-14, 22
+    
+With these changes, the image viewer will now use native dialogs on most platforms. The platforms supported are iOS, Linux (with a GTK+ platform theme), macOS, Windows and WinRT. For Android, it will use a default Qt dialog provided by the ``QtWidgets`` module.
+    
+Common Patterns
+===============
+
+There a number of common user interface patterns that can be implemented using Qt Quick Controls 2. In this section, we try to demonstrate how some of the more common ones can be built.
+
+Nested Screens
+--------------
+
+For this example we will create a tree of pages that can be reached from the previous level of screens. The structure is pictured below.
+
+.. digraph:: nested_screens
+
+    "Home" -> "Profile"
+    "Profile" -> "Edit Profile"
+    "Home" -> "About"
+
+The key component in this type of user interface is the ``StackView``. It allows us to place pages on a stack which then can be poped when the user wants to go back. In the example here, we will show how this can be implemented.
+
+The initial home screen of the application is shown in the figure below.
+
+.. figure:: assets/interface-stack-home.png
+
+    The home screen page.
+
+The application start in ``main.qml`` where we have an ``ApplicationWindow`` containing a ``ToolBar``, a ``Drawer``, a ``StackView`` and a home page element, ``Home``. We will look into each of the components below.
+
+.. literalinclude:: src/interface-stack/main.qml
+    :lines: 1-4, 10-16, 36-41, 66-72, 84
+
+The home page, ``Home.qml`` consists of a ``Page``, which is an control element that support headers and footers. In this example we simply center a ``Label`` with the text *Home Screen* on the page.
+
+.. literalinclude:: src/interface-stack/Home.qml
```

Comment:
It could be worth mentioning that each item in a StackView fills the StackView by default, which is why we're able to centre the label here.